src/main/drivers/timer.h

```diff
@@ -107,6 +111,26 @@ typedef struct timerHardware_s {
 #endif
 } timerHardware_t;
 
+#ifdef USE_TIMER_MGMT
+
+typedef uint8_t timerTag_t;
+
+typedef struct timerChannelConfig_s {
```

Comment:
I'm afraid it would be support nightmare. Setting wrong values can cause 'interesting' things.
Flash use should not be that bad
And most of the tables is ready (in reprocessor checks)